src/hotspot/cpu/x86/globals_x86.hpp

```diff
@@ -44,7 +44,7 @@ define_pd_global(uintx, CodeCacheSegmentSize,    64 COMPILER1_AND_COMPILER2_PRES
 // the uep and the vep doesn't get real alignment but just slops on by
 // only assured that the entry instruction meets the 5 byte size requirement.
 #if COMPILER2_OR_JVMCI
-define_pd_global(intx, CodeEntryAlignment,       32);
+define_pd_global(intx, CodeEntryAlignment,       64);
```

Comment:
How about setting dynamic alignment based on MaxVectorSize ? i.e. match the alignment to MaxVectorSize. This way we can save internal fragmentations over AVX2.
It could be done during VM Initialization similar to following code.
https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/vm_version_x86.cpp#L1439